helical/models/caduceus/model.py

```diff
@@ -0,0 +1,151 @@
+    We make use of the implementation from the [Caduceus](https://github.com/kuleshov-group/caduceus) repository.
+    """
+    def __init__(self, configurer: CaduceusConfig = configurer):
+        super().__init__()
+
+        if torch.cuda.is_available() == False:
+            message = "Caduceus requires a CUDA device to run and CUDA is not available"
+            LOGGER.error(message)
+            raise RuntimeError(message)
+            
+        self.configurer = configurer
+        self.config = configurer.config
+        self.files_config = configurer.files_config
+        self.device = self.config['device']
+
+        downloader = Downloader()
+        for file in self.configurer.list_of_files_to_download:
+            downloader.download_via_name(file)
+
+        self.model =  CaduceusModel.from_pretrained(self.files_config['model_files_dir'])
+        self.model.eval()
+
+        self.tokenizer = CaduceusTokenizer(model_max_length=self.config['input_size'])
+        
+        LOGGER.info("Caduceus model initialized")
+
+    def _collate_fn(self, batch):
+        input_ids = torch.tensor([item["input_ids"] for item in batch]).squeeze(1)
+        batch_dict = {
+            "input_ids": input_ids,
+        }
+
+        if "labels" in batch[0]:
+            batch_dict["labels"] = torch.tensor([item["labels"] for item in batch])
+            
+        return batch_dict
+    
+    def _pool_hidden_states(self, hidden_states, sequence_length_dim=1):
+        """Pools hidden states along sequence length dimension."""
+        if self.config["pooling_strategy"] == "mean":  # Mean pooling along sequence length dimension
+            return hidden_states.mean(dim=sequence_length_dim)
+        if self.config["pooling_strategy"] == "max":  # Max pooling along sequence length dimension
+            return hidden_states.max(dim=sequence_length_dim).values
+        if self.config["pooling_strategy"] == "last":  # Use embedding of last token in the sequence
+            return hidden_states.moveaxis(hidden_states, sequence_length_dim, 0)[-1, ...]
+        if self.config["pooling_strategy"] == "first":  # Use embedding of first token in the sequence
+            return hidden_states.moveaxis(hidden_states, sequence_length_dim, 0)[0, ...]
+
+    def process_data(self, sequences: List[str]):
+        """Process the input DNA sequences.
+
+        Parameters 
+        ----------
+        sequences : list[str]
+            The input DNA sequences to be processed.
+
+        Returns
+        ----------
+        HyenaDNADataset
+            Containing processed DNA sequences.
+
+        """
+        LOGGER.info("Processing data")
+
+        self.ensure_dna_sequence_validity(sequences)
+        
+        tokenized_sequences = []
+        for seq in sequences:
+            tokenized_seq = self.tokenizer(seq, return_tensors="pt", padding="max_length", truncation=True, max_length=self.config['input_size'])
```

Comment:
make these parameters available to be set in the function `process_data()` above so that they could be modified.